### ISSUE TITLE
Checks on observables in custom likelihoods

### ISSUE THREAD

**Issue by MJKirk** [smelli/smelli#43]
At the moment, if you add a custom likelihood there are no checks  on the observables you pass, in particular to check if 
a) They exist in flavio, or
b) They are already being included in the global likelihood fit instance.
As an example, if one does:
```python
gl = smelli.GlobalLikelihood(include_likelihoods = ("likelihood_ewpt.yaml",), 
                             custom_likelihoods = {"bsgamma": ["BR(B->Xsgamma)", "BR(B0->K*gamma)"],
                                                   "sometypos": ["AFB(Z->ee)", "AFB(Z-mumu)", "AFB(Z-tautau)"]}
)
```
you don't get any errors, but the "custom_bsgamma" likelihood just gives 0, and the "custom_sometypos" gives something, but not actually what we want.

This is because the custom likelihood code just looks at the difference between full set of observables and the given custom list, and then gets the likelihood excluding that difference, but never checks the list it is given.
https://github.com/smelli/smelli/blob/550e2016f2f6042757bac4262b3c441705911cf7/smelli/classes.py#L760-L787

I would propose to add some checks to throw an error, or at least a warning, in this case.
Something like:
```python
all_obs = set()
for lh in gl.likelihoods.values():
  all_obs.add(set(lh.observables)
for flh in gl.fast_likelihoods.values():
  all_obs.add(set(flh.observables))
if not set(self.observables).issubset(all_obs):
  raise ValueError
```
in the CustomLikelihood class constructor would do the job. (I just wrote this straight into github, so consider it approximate pseudo-code)

**Comment by peterstangl** [comment]
Thanks for this suggestion! I have implemented some new checks in PR #44. Could you please take a look and see if this does fully address the issue?

**Comment by peterstangl** [comment]
Implemented in PR #44.
